source/write/run-command.txt

```diff
@@ -0,0 +1,180 @@
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: administration, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+php-library+}
+to run a database command. You can use database commands to perform a
+variety of administrative and diagnostic tasks, such as fetching server
+statistics, initializing a replica set, or running an aggregation pipeline.
+
+.. important:: Prefer Library Methods to Database Commands
+
+   The library provides wrapper methods for many database commands.
+   We recommend using these methods instead of executing database
+   commands when possible.
+   
+   To perform administrative tasks, use the :mongosh:`MongoDB Shell </>`
+   instead of the {+php-library+}. Calling the ``db.runCommand()``
+   method inside the shell is the preferred way to issue database
+   commands, as it provides a consistent interface between the shell and
+   drivers or libraries.
+
+.. _php-execute-command:
+
+Execute a Command
+-----------------
+
+To run a database command, you must specify the command and any relevant
+parameters in a command document, then pass the command document to the
+``MongoDB\Database::command()`` method. This method returns a
+``Cursor`` object.
+
+The following code shows how you can use the ``command()``
+method on a ``MongoDB\Driver\Database`` instance to run the ``hello``
+command, which returns information about the current member's role in
+the replica set:
+
+.. code-block:: php
+
+   $database = $client->selectDatabase('<db>');
+   $result = $database->command(['hello' => 1]);
+
+To find a link to a full list of database commands and corresponding
+parameters, see the :ref:`Additional Information section
+<php-addtl-info-runcommand>`.
+
+.. note:: Read Preference
+
+   The ``command()`` method does not obey the read preference you might
+   have set on your ``Database`` instance elsewhere in your code. By
+   default, ``command()`` uses the ``primary`` read preference.
+   
+   You can set a read preference for command execution by setting one
+   in the options parameter, as shown in the following code:
+   
+   .. code-block:: php
+
+      $readPref = new MongoDB\Driver\ReadPreference('primaryPreferred');
+      $result = $database->command(
+          ['hello' => 1], 
+          ['readPreference' => $readPref]
+      );
+
+   For more information on read preference options, see :manual:`Read
+   Preference </core/read-preference/>` in the {+mdb-server+} manual.
+
+.. _php-command-response:
+
+Response
+--------
+
+The ``command()`` method returns a ``Cursor`` object that contains
+the response from the database after the command has been executed.
+
+Each database command performs a different function, so the response
+content can vary depending on the command executed. However, every
+response contains a document with the following fields:
```

Comment:
For commands that return a cursor (e.g. `aggregate`), users are never going to have access to these fields because the driver converts the cursor envelope (e.g. ID and first batch of results) in the raw command response into an iterable cursor. This differs from commands that return a single document, which the driver would package as a cursor object containing a single result (which is why we use `$cursor->toArray()[0]`).

If you're going to talk about how "response content can vary", I would clearly explain this difference between commands and mention how the PHP driver handles this. So users are responsible for knowing the response format of their command (usually discussed in the server manual) so they can either iterate or just dump the first and only document in the cursor.

For an example of a cursor envelop in a command response, see the first output example in [listCollections](https://www.mongodb.com/docs/manual/reference/command/listCollections/#list-all-collections). Running `listCollections` through PHP's `command()` helper will never reveal that raw response, as we're going to construct a cursor from the first batch. There will also be no more results, since the cursor ID is zero (i.e. all results returned in the first batch). If the cursor ID was non-zero, then iterating beyond the first batch would utilize a `getMore` command to read more cursor data from the server.

I just noticed there's some confusion about this in the server manual and opened [DOCSP-43843](https://jira.mongodb.org/browse/DOCSP-43843) to report that.

Reply:
I'll revise the introduction to that table, I will make the language more clear and indicate what the library behavior is here!